renderer/components/Profile/ProfilePaneSignMessage/ProfilePaneSignMessage.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useRef } from 'react'
+import PropTypes from 'prop-types'
+import { FormattedMessage, injectIntl, intlShape } from 'react-intl'
+import { Box, Flex } from 'rebass'
+import { grpc } from 'workers'
+import { Bar, CopyBox, TextArea, Text, Form, Button } from 'components/UI'
+import messages from './messages'
+
+const ProfilePaneNodeInfo = ({ intl, showNotification, ...rest }) => {
+  const [sig, setSig] = useState(null)
+
+  const formApiRef = useRef(null)
+  const notifyOfCopy = () =>
+    showNotification(intl.formatMessage({ ...messages.sig_copied_notification_description }))
+
+  const onSignMessage = async () => {
+    try {
+      const { current: formApi } = formApiRef
+      const message = formApi.getValue('message')
+      const { signature } = await grpc.services.Lightning.signMessage({ msg: Buffer.from(message) })
```

Comment:
Should we extract this to a redux thunk rather than using the service directly in the component to abstract the service api specifics out of the component and be more consistent with all other service invocations?

Reply:
I'm a bit skeptical of this. That action won't do anything. Not even dispatch and IPC event or do anything that can affect something in another part of the app. Feels kinda boilerplaty or consistency for the sake of consistency. But if you insist I can add that.

Reply:
There are a couple of advantages to doing it this way.

1) Better testability by leveraging dependency injection
2) Better component reusability for the same reason
3) Easier debugging by ensuring a consistent approach to logging (all actions logged the same way)
4) Hiding API complexity from our components (using Buffers in this case)
5) General code consistency

I don't see that there is lots of boilerplate code to do this - it's just defining and exporting a wrapper thunk and injecting it into the component with connect.

Reply:
1. Agree
2. Partly agree. Theoretically maybe practically in this concrete case - not so much
3. Agree but with some remarks because of pt4
4. Kinda. But leveraging redux here doesn't feel right. One can hide complexity by adding some sort of a util/wrapper. There are other ways to do dep injections too.
5. Consistency is good, but not for _just_ for the sake of it.

Reply:
> Partly agree. Theoretically maybe practically in this concrete case - not so much

I was thinking specifically about wanting to display this component in the storybook without having to build a general mock for the grpc worker. Mocking a redux action that is injected via a prop is probably simpler.

> Kinda. But leveraging redux here doesn't feel right. 

In what way does redux not feel right? Because we may not be mutating state off the back of the action?

> One can hide complexity by adding some sort of a util/wrapper. There are other ways to do dep injections too.

Such as what? Only way I know to inject things into components is via props. Perhaps the prop doesn't have to be a redux action. It could be some other wrapper. But, in the context of our codebase I don't know of a more suitable place to wrap things like api calls than our reducers.

> Consistency is good, but not just for the sake of it.

I agree. If there were no other gains to be had I would be less inclined to change the implementation just for the sake of consistency.

Reply:
>Mocking a redux action that is injected via a prop is probably simpler.
You can mock a prop
>In what way does redux not feel right? Because we may not be mutating state off the back of the action?
It doesn't affect the state in any way and we know that before and are not planning that it will. 
>Such as what?

Reply:
>Mocking a redux action that is injected via a prop is probably simpler.

You can mock a prop

>In what way does redux not feel right? Because we may not be mutating state off the back of the action?

It doesn't affect the state in any way and we know that before and are not planning that it will. 

>Such as what?

Injecting a prop doesn't mean it has to be a redux action. You can use HOC, hook, render props, pass props directly after all.

As far as logging is concerned - all GRPC calls are also logged to a console

PS: as i said in the first post of this chain - if you want I can make this change, it's not something that i consider a no go no matter what, just thinking that it may be sub optimal in certain ways. This convo is more to discuss what are the pros and the cons and what are the other ways

Reply:
> It doesn't affect the state in any way and we know that before and are not planning that it will.

We use the redux store to describe the current state of the app so that components or any other part of the app has the ability to act on that in one way or another. IMO The fact that we are choosing not to act directly on that at this point is neither here nor there - it's still good to have the store accurately reflect what's happening in the app at any given time.

> PS: as i said in the first post of this chain - if you want I can make this change, it's not something that i consider a no go no matter what, just thinking that it may be sub optimal in certain ways. This convo is more to discuss what are the pros and the cons and what are the other ways

I know, I get that. I'm just not keen on the idea of components importing and using services directly as it feels like a rather tight coupling of a component to the mechanics of a specific API call, and I'm trying to understand the objection to using redux for this as we do everywhere, and wether there is another approach that we could be taking.